src/Form/index.tsx

```diff
@@ -1079,11 +1065,11 @@ function Form({
       submitData || ['button', 'text'].includes(metadata.elementType);
     if (!redirectKey) {
       if (explicitNav) {
-        setFormCompleted(true);
         eventData.completed = true;
-        if (internalState[_internalId])
-          internalState[_internalId].formCompleted = true;
-        authHookCb.cb();
+        initState.formSessions[formName].form_completed = true;
```

Comment:
maybe get the session at the top of the component so it can be used directly where needed

const session = initState.formSessions[formName];
...
session.form_completed = true;
...
session.form_completed && ...